### DISCUSSION TITLE
Solar radiation budget problem in NoahMP LSM

### DISCUSSION THREAD

**Discussion by OBJ780** [NCAR/hrldas#180] [Q&A]
Q1: When I ran the Noah-MP single-point model, I attempted to modify the options in the DYNAMIC_VEG_OPTION module to 1, 6, and 7, which include the parameterization options with "use FVEG = SHDFAC from input." However, the model encountered radiation balance issues. I am not quite clear about what the SHDFAC parameter represents and where I can find this parameter？

Q2: When I simulated heat flux using the default parameterization scheme and other schemes without the "use FVEG = SHDFAC from input" option, the model results were reasonably close to the observed values before 10 a.m. However, there was a significant underestimation after 10 a.m., which I find unreasonable. I'm not sure what the reason is and whether it is related to the radiation balance issues encountered earlier？I will upload the parameter settings of my driver file and simulation result plots.
![image](https://github.com/NCAR/hrldas/assets/156399638/369fb901-faf3-421d-b641-4c053427f7ca)
![image](https://github.com/NCAR/hrldas/assets/156399638/53124f51-5307-4026-970e-0155005e0c6a)
![image](https://github.com/NCAR/hrldas/assets/156399638/e30df94f-75cf-4d75-99af-f9d8dc905324)
![image](https://github.com/NCAR/hrldas/assets/156399638/3e991193-a026-425d-a8c6-4ba167634c17)
![image](https://github.com/NCAR/hrldas/assets/156399638/049aa9c8-3554-4493-85fe-d0605390a578)

**Comment by RAbolafiaRosenzweig** [comment]
When using a Noah-MP physics option that notes: "use FVEG = SHDFAC from input.", the vegetation fraction (i.e., SHDFAC) is read from the forcing files and thus needs to be added as an additional variable to these files. Time series of vegetation green fraction can be derived from satellite observations of green vegetation fraction.

RE diurnal cycle errors: My hypothesis is this mismatch may be due to forcing-related issues or a time-zone mismatch. Please check: forcing data (and in turn simulation data) and the observational data are using the same time zone and the forcing radiation data provides a reasonable diurnal cycle that is consistent with observations.

**Reply by OBJ780** [reply]
Thank you for your suggestions. Your guess is correct; I did not use UTC time, which led to the simulation results looking unreasonable. However, I am not sure how to add SHDFAC as an additional variable to my driver file. I noticed there are maximum_vegetation_pct and minimum_vegetation_pct in the .dat file; should I modify them here? What data product should I download to obtain vegetation fraction data? I tried modifying the LAI values in the .dat file, as well as the values of maximum_vegetation_pct and minimum_vegetation_pct, but when I choose DYNAMIC_VEG_OPTION=1 or 7, the model still encounters radiation balance problems, and only outputs a few timestep files before stopping.

**Reply by cenlinhe** [reply]
please test using DYNAMIC_VEG_OPTION=4 to see if it works. If it still does not work, that means your atmospheric forcing data may have some incorrect/unreasonable values.

**Reply by OBJ780** [reply]
When using DYNAMIC_VEG_OPTION=4, the model functions properly. However, when using DYNAMIC_VEG_OPTION=1 or 7, the model encounters a stop error: "Solar radiation budget problem in NoahMP LSM", and only outputs nc files up to the time step 2016 06 30 22 00. Below is the driver code I'm using with UTC time.
![image](https://github.com/NCAR/hrldas/assets/156399638/8faccd1e-408a-4225-a31a-c0ee70a0c0e2)

**Reply by cenlinhe** [reply]
This is because you need to provide vegetation fraction variable in your atmos forcing file to use DYNAMIC_VEG_OPTION=1 or 7. Please add a variable named "VEGFRA" in your forcing file together with variables like SWDOWN, T2D, V2D, ... if you want to use DYNAMIC_VEG_OPTION=1 or 7. The "VEGFRA" variable is the vegetation fraction timeseries for each grid, which can be obtained from satellite data (e.g., https://www.ospo.noaa.gov/Products/land/gvf/) with some temporal interpolation.

**Reply by OBJ780** [reply]
In Noah-MP single-point mode with DYNAMIC_VEG_OPTION=1 or 7, should I add variables like VEGFRA, SWDOWN, T2D, and V2D in the Initial States section similar to input LAI? Or should I provide variable values for each time step, similar to shortwave and longwave?

**Reply by cenlinhe** [reply]
I would suggest adding VEGFRA directly to the generated netcdf forcing file *LDASIN_DOMAIN for each file (i.e., each time step).

**Reply by OBJ780** [reply]
OK，I get  it. Thank you for your help, you've helped me solve some tricky problems.

**Comment by cenlinhe** [comment]
Note that Forcing data has to be on UTC time not local time.

**Reply by OBJ780** [reply]
Yes, you're right. I'm using Beijing time, and when I convert it to UTC, the simulated values of heat flux seem closer to the observed values. Thank you for your response！
